### DISCUSSION TITLE
How am I supposed to use/process the parameter block at the end of Cura G-code files ???

### DISCUSSION THREAD

**Discussion by PhilBaz** [smartavionics/Cura#169] [General]
Is there a way I can use this to generate a profile, or at least transform it into something non-mind-bending to try and read. For fun the names don't match the parameters in the UI either....

e.g. 

;End of Gcode
;SETTING_3 {"global_quality": "[general]\\nversion = 4\\nname = 2.4 PLA_Vase_22-
;SETTING_3 02-10\\ndefinition = voron2_350\\n\\n[metadata]\\ntype = quality_chan
;SETTING_3 ges\\nquality_type = fast\\nsetting_version = 20\\n\\n[values]\\nacce
;SETTING_3 leration_enabled = True\\nadhesion_type = brim\\nbridge_settings_enab
;SETTING_3 led = False\\nlayer_height_0 = 0.3\\nmagic_spiralize = True\\nmateria
;SETTING_3 l_bed_temperature = 60\\nmaterial_bed_temperature_layer_0 = 60.0\\nre
;SETTING_3 traction_combing = noskin\\nspeed_slowdown_layers = 0\\ntravel_retrac
;SETTING_3 t_before_outer_wall = True\\n\\n", "extruder_quality": ["[general]\\n
;SETTING_3 version = 4\\nname = 2.4 PLA_Vase_22-02-10\\ndefinition = voron2_350\
;SETTING_3 \n\\n[metadata]\\ntype = quality_changes\\nquality_type = fast\\ninte
;SETTING_3 nt_category = default\\nposition = 0\\nsetting_version = 20\\n\\n[val
;SETTING_3 ues]\\nacceleration_print = 7000.0\\nacceleration_topbottom = 3500.0\
;SETTING_3 \nacceleration_travel = 7000.0\\nacceleration_wall_x = 3500\\nalterna
;SETTING_3 te_extra_perimeter = False\\nbottom_layers = 2\\nbridge_enable_more_l
;SETTING_3 ayers = False\\nbrim_line_count = 12\\ncool_fan_full_layer = 6\\ncool
;SETTING_3 _fan_speed_0 = 8\\ncool_fan_speed_max = 30\\ncool_fan_speed_min = 30\
;SETTING_3 \ncool_min_layer_time = 0.01\\ncool_min_layer_time_fan_speed_max = 10
;SETTING_3 \\nfill_outline_gaps = False\\ninfill_enable_travel_optimization = Fa
;SETTING_3 lse\\ninfill_overlap = 25\\ninfill_pattern = cubicsubdiv\\ninfill_spa
;SETTING_3 rse_density = 30\\ninfill_wipe_dist = 0\\ninitial_bottom_layers = 0\\
;SETTING_3 nline_width = 0.5\\nmaterial_print_temperature = 210.0\\nmaterial_pri
;SETTING_3 nt_temperature_layer_0 = 210.0\\nmin_infill_area = 0\\nretract_at_lay
;SETTING_3 er_change = True\\nretraction_amount = 0.5\\nretraction_hop_enabled =
;SETTING_3  True\\nretraction_prime_speed = 18\\nretraction_speed = 18\\nroofing
;SETTING_3 _layer_count = 0\\nskin_monotonic = True\\nskin_overlap = 25\\nskirt_
;SETTING_3 brim_speed = 60\\nskirt_gap = 10\\nskirt_line_count = 2\\nspeed_wall_
;SETTING_3 0 = 60\\nsub_div_rad_add = 0\\nsupport_brim_enable = True\\nsupport_x
;SETTING_3 y_distance = 0.4\\nsupport_z_distance = 0.2\\ntop_layers = 0\\ntravel
;SETTING_3 _avoid_other_parts = True\\ntravel_avoid_supports = True\\nwall_0_mat
;SETTING_3 erial_flow = =wall_material_flow\\nwall_0_wipe_dist = 0.2\\nwall_line
;SETTING_3 _count = 4\\nwall_line_width = 0.5\\nwall_overhang_angle = 80\\nxy_of
;SETTING_3 fset_layer_0 = 0\\nz_seam_corner = z_seam_corner_weighted\\nz_seam_ty
;SETTING_3 pe = back\\n\\n"]}

**Comment by smartavionics** [comment]
I should think the easiest way to transform that into something more readable would be to write a little Python program to join those lines together and strip out the prefixes and convert the \n sequences into line breaks.

**Reply by PhilBaz** [reply]
Ahhh, okay....that might be a nice little learning exercise for me. I can break it apart in excel to read a little nicer. But I probably should learn some rudimentary python.
